Журнал тестов старт 12.01.2025 в 0:20

Тестирование 12.01.2025 0:37
Версия 1.6a0.
Тестирование производится на Windows 11 x64 и python 3.13.1 x64
Тестирование функций exchange_server и exchange_client из diffiehellmanlib.exchange
Bugs list:
    - Bug 1:
        - Traceback (most recent call last):
            File "C:\Users\kostya\Desktop\tests\server.py", line 3, in <module>
                shared_key = exchange.exchange_server("0.0.0.0", 5000, "127.0.0.1")
            File "C:\Users\kostya\Desktop\tests\venv\Lib\site-packages\diffiehellmanlib\exchange\exchange.py", line 79, in exchange_server
                shared_key = wrapper.generate_shared_key(B, p, g, a)
            File "C:\Users\kostya\Desktop\tests\venv\Lib\site-packages\diffiehellmanlib\wrapper.py", line 120, in generate_shared_key
                return hash_shared_key(shared_key_value)
            File "C:\Users\kostya\Desktop\tests\venv\Lib\site-packages\diffiehellmanlib\wrapper.py", line 106, in hash_shared_key
                dif_helm.hash_shared_key(shared_key.encode(), ctypes.byref(hashed_key))
                                         ^^^^^^^^^^^^^^^^^
            AttributeError: 'bytes' object has no attribute 'encode'. Did you mean: 'decode'?
    Fix:
       - wrapper.py 
           - replace dif_helm.hash_shared_key(shared_key.encode(), ctypes.byref(hashed_key)) to dif_helm.hash_shared_key(shared_key, ctypes.byref(hashed_key))
    Status : FIXED in 1.6a1

Примечание: Пока для теста я у себя библиотеке поменял строчку чтобы дозавершить тест.
Тест завершён.
Баги не обнаружены.

Результат сессии теста.
Что нужно поменять:
    - wrapper.py:
        - строка 106 с dif_helm.hash_shared_key(shared_key.encode(), ctypes.byref(hashed_key)) на dif_helm.hash_shared_key(shared_key, ctypes.byref(hashed_key))
Остальные проверку прошли.
Конец записи тестирования.
